tools/sim/bridge.py

```diff
@@ -54,17 +57,65 @@ def steer_rate_limit(old, new):
     return new
 
 frame_id = 0
+vipc_server = None
+ctx = None
+queue = None
+krnl = None
+Wdiv4 = 0 
+Hdiv4 = 0 
 def cam_callback(image):
   global frame_id
+  global vipc_server
```

Comment:
this is way too many globals. let's put all this stuff in a camerad class

Reply:
My use of the globals was to prevent redundant recalculations and allocation/deallocation on the repeated callbacks, particularly on the GPU that may not handle it that well. I'm not very familiar with Python. Can you offer an example of how to avoid the globals and achieve the same result? Or are you simply saying that you want to "bundle them up" in a single global class?

Reply:
yeah, just wrap all this in a class

Reply:
I haven't studied performance very carefully, but it seems to consume pretty neglible gpu and cpu on my machine.  No problem keeping up at 20 Hz. I will try to characterize it a bit better.

Reply:
cool, main concerns is is it doing 20Hz comfortably on modest hardware?

Reply:
I have a pretty powerful machine, 8 core Alienware area51 with rtx2070.  I'll try setting up some "torture testing", perhaps by repeating the whole process N times and see where it breaks down.

Reply:
Camerad class is incorporated now. I took some performance measurements, and they look pretty good. I iterated the yuv conversion routine N times and observed the total bridge.py CPU usage, where 12.5 % is one fully occupied CPU core. I did not repeat any of the I/O. When I got up to 14 iterations some rare "slow frame" events happened. Hardware able to run Carla should easily work.

       N   %CPU
  	1	5
  	3	6
 	5	7
 	7	8
 	8	9
 	9	9
 	10	10
 	11	10
 	12	11
 	13	12
 	14	12 (very rare slow frame)
 	15	12 (rare slow frame)
 	16    12 (some slow frames)